fennel.lua

```diff
@@ -1461,6 +1461,8 @@ docSpecial('.', {'tbl', 'key1', '...'},
 
 SPECIALS['global'] = function(ast, scope, parent)
     assertCompile(#ast == 3, "expected name and value", ast)
+    assertCompile(not scope.unmanglings[tostring(ast[2])],
+                  "global " .. tostring(ast[2]) .. " conflicts with local", ast)
```

Comment:
This looks like a good approach, but it won't catch it if there's destructuring in play. For example,
```clojure
(local a-b 1)
(global [a_b] [2])
```
bypasses the check.

Reply:
Nice catch! Updated to handle destructuring as well as do extra mangling in the reverse case as per the comment in #197.

Reply:
Nice, I thought addressing destructuring was going to end up messier - *and* it looks like you ended up addressing another unfiled bug in the process with the destructuring + global tracking!